PetaPoco/Utilities/EnumMapper.cs

```diff
@@ -23,7 +23,15 @@ public static object EnumFromString(Type enumType, string value)
                 return newmap;
             });
 
-            return map[value];
+            try
+            {
+                return map[value];
+            }
+            catch (KeyNotFoundException)
+            {
+                throw new InvalidOperationException(
+                    string.Format("Can not convert value `{0}` into {1}-Enum", value, enumType.Name));
```

Comment:
If I call `Enum.Parse()` with a bad value, I get an `ArgumentException` with message `Requested value 'foo' was not found.` I think the exception and message should be the same here -- or possibly append the type name as well.

```csharp
throw new ArgumentException($"Requested value '{value}' was not found in enum {enumType.Name}.");
```

(Note also that technically this is a breaking change. Maybe a compromise would be to keep the `KeyNotFoundException` and just change the message?)

Reply:
I like it. (I really didn't like the `InvalidOperationException` because that's not what it is.. I would have loved to have a `EitherYourDatabaseIsCorruptedOrYourPocosAreOutOfDateException` but one can't have it all, can one?)

I'll go with the `KeyNotFoundException` - that fits just as well and avoids the breaking change..